src/main/java/com/epam/javacourses2016/task1/SolverTask1.java

```diff
@@ -16,7 +20,32 @@
      * @return Список строк, прочитанных из входного файла в прямом порядке.
      */
     public List<String> reverseFile(File input, File output) {
-        //TODO: realize solver of a task
-        return null;
+        List<String> list = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(input)))
+        {
+            String sCurrentLine;
+            while ((sCurrentLine = br.readLine()) != null) {
+                list.add(sCurrentLine);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        List<String> reversedList = new ArrayList<>(list);
+        Collections.reverse(reversedList);
+        try {
+            if (!output.exists()) {
+                output.createNewFile();
+            }
+            FileWriter fw = new FileWriter(output.getAbsoluteFile());
```

Comment:
Why don't you use try-with-resource on Writer? It should also be closed after use